apps/prairielearn/elements/pl-excalidraw/pl-excalidraw.py

```diff
@@ -0,0 +1,133 @@
+import base64
+import json
+from pathlib import Path
+
+import chevron
+import lxml.html
+import prairielearn as pl
+from lxml.html import HtmlElement
+from typing_extensions import assert_never
+
+ATTR_ANSWER_NAME = "answers-name"
+ATTR_WIDTH = "width"
+ATTR_HEIGHT = "height"
+ATTR_SOURCE_FILE_NAME = "source-file-name"
+ATTR_SOURCE_DIRECTORY = "directory"
+
+
+SOURCE_DIRECTORY_MAP = {
+    "serverFilesCourse": "server_files_course_path",
+    "clientFilesCourse": "client_files_course_path",
+    "clientFilesQuestion": "client_files_question_path",
+    "courseExtensions": "course_extensions_path",
+    ".": "question_path",
+}
+
+
+def prepare(element_html: str, data: pl.QuestionData) -> None:
+    element = lxml.html.fragment_fromstring(element_html)
+    required_attrs = [ATTR_ANSWER_NAME]
+    optional_attrs = [
+        ATTR_WIDTH,
+        ATTR_HEIGHT,
+        ATTR_SOURCE_FILE_NAME,
+        ATTR_SOURCE_DIRECTORY,
+    ]
+    pl.check_attribs(element, required_attrs, optional_attrs)
+
+    name = pl.get_string_attrib(element, ATTR_ANSWER_NAME)
+    pl.check_answers_names(data, name)
+
+    source_dir = pl.get_string_attrib(element, ATTR_SOURCE_DIRECTORY, ".")
+    if source_dir not in SOURCE_DIRECTORY_MAP:
+        raise RuntimeError(
+            f"{source_dir=} has to be one of {list(SOURCE_DIRECTORY_MAP.keys())}"
```

Comment:
Nit, for consistency with language used in all other elements:

```suggestion
            f"{source_dir=} must be one of {list(SOURCE_DIRECTORY_MAP.keys())}"
```